lib/Models/Cesium.js

```diff
@@ -423,6 +423,138 @@ Cesium.prototype.computePositionOnScreen = function(position, result) {
     }
 };
 
+/**
+ * Updates the order of layers on the Leaflet map to match the order in the Now Viewing pane.
+ */
+Cesium.prototype.updateLayerOrder = function() {
+    // move alwaysOnTop layers to the top
+    var items = this.terria.nowViewing.items;
+    var scene = this.scene;
+    for (var l = items.length - 1; l >= 0; l-- ) {
+        if (items[l].imageryLayer && items[l].keepOnTop) {
+            scene.imageryLayers.raiseToTop(items[l].imageryLayer);
+        }
+    }
+};
+
+Cesium.prototype.updateLayerOrderAfterReorder = function() {
+    // because this Cesium model does the reordering via raise and lower, no action needed.
+};
+
+// useful for counting the number of items in composite and non-composite items
+function numSubItems(item) {
+    var N = 1;
+    if (defined(item.numItems)) {
+        N = item.numItems();
```

Comment:
I think I would just use `item.items.length` here, instead of this `numItems` function.  Because:
* This code already needs to know about sub-items and how to manage them, so there's little reason it shouldn't also know how to count them.
* `numItems` is a bit general for the public API.  Based on the name, it sounds like it should also be implemented on `CatalogGroup`.